### PR TITLE
refactor: make `versions` package usable locally

### LINKED ISSUE #849: Make `versions` package usable locally
Currently, the `versions` package returns only `0.0.0` when developing. Then, the CI injects the proper versions during the release routine so that the published package will have all proper versions injected into it. Although this works fine, it's not enough.

We need to ensure that these versions are **_also_** set for local development _(instead of `0.0.0`)_, so they can be used by other packages and scripts that rely on this information to perform other actions.

> **Note** _Check [this](https://github.com/FuelLabs/fuels-ts/pull/811#discussion_r1142747647) for more insights on the matter._

### PR DESCRIPTION
**Pull request by arboleya**
- Close #849

### CODE REVIEW

File: scripts/changeset-version-with-docs.ts [hunk 1]
```diff
@@ -6,6 +6,10 @@ import sh from 'shelljs';
   // Force exit on error
   sh.set(`-e`);
 
+  // Commit versions generated at pre-build step
+  sh.exec(`git add packages/versions/*`);
+  sh.exec(`git commit -m"ci(scripts): update versions"`);
+
```
Comment (arboleya) [inline scripts/changeset-version-with-docs.ts:12]: @camsjams The `versions` package now has a `prebuild` script that will update a `.ts` file, injecting them from the CI env vars, similar to how it is done for `versions.yml`.

I want to commit these changes from the CI, so the `FUELS` version is always up to date.

With this, we can use this package <sup>_(after #827 goes to `master`)_</sup> on the new doc engine.

Ref: https://github.com/FuelLabs/fuels-ts/pull/811#discussion_r1141633562

> cc @Torres-ssf

File: packages/versions/scripts/rewriteVersions.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+export const readVersions = () => {
+  const fromFiles = readVersionsFromFiles();
+  const fromEnv = readVersionsFromEnv();
+
+  const FUELS = fromEnv.FUELS || fromFiles.FUELS;
+  const FORC = fromEnv.FORC || fromFiles.FORC;
+  const FUEL_CORE = fromEnv.FUEL_CORE || fromFiles.FUEL_CORE;
+
+  return { FUELS, FORC, FUEL_CORE };
+};
```
Comment (Torres-ssf) [inline packages/versions/scripts/rewriteVersions.ts:50]: :ok_hand:

File: packages/versions/scripts/rewriteVersions.ts [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  const filepath = join(__dirname, '..', 'src', 'lib', 'getSupportedVersions.ts');
+
+  let contents = readFileSync(filepath, 'utf8');
+
+  contents = contents.replace(/FUELS: '[\d.]+'/, `FUELS: '${FUELS}'`);
+  contents = contents.replace(/FORC: '[\d.]+'/, `FORC: '${FORC}'`);
+  contents = contents.replace(/FUEL_CORE: '[\d.]+'/, `FUEL_CORE: '${FUEL_CORE}'`);
+
+  writeFileSync(filepath, contents);
+};
```
Comment (Torres-ssf) [inline packages/versions/scripts/rewriteVersions.ts:64]: Awesome :rocket:

**Review by Torres-ssf — APPROVED** [review]

**Review by Dhaiwat10 — APPROVED** [review]
Nicely done 🪄

**Pull request merged**